components/event-system/queue.js

```diff
@@ -0,0 +1,89 @@
+const HOW_MANY_PRIORITIES_ARE_THERE_NITRO = 6
+const eventQueue = Array.from(Array(HOW_MANY_PRIORITIES_ARE_THERE_NITRO), () => [])
```

Comment:
I don't like (again) that this is global. With this setup, we can only have one instance of `eventQueue`. I'd prefer we rewrite this as a class that must be instantiated so that we can have multiple queues.